server/sonar-server/src/main/java/org/sonar/server/edition/MutableEditionManagementState.java

```diff
@@ -63,6 +63,15 @@
   PendingStatus automaticInstallReady();
 
   /**
+   * Uninstalls the currently installed edition
+   *
+   * @return the new pending status, always {@link PendingStatus#UNINSTALL_IN_PROGRESS}
+   *
+   * @throws IllegalStateException if current status is not {@link PendingStatus#NONE}
+   */
+  PendingStatus uninstall();
```

Comment:
uninstall should fail with "IllegalStateException" if there is no current edition

Reply:
ok